kubessh/pod.py

```diff
@@ -200,6 +225,19 @@ async def ensure_running(self):
         if not pod:
             # There is no pod, so start one!
             yield PodState.STARTING
+
+            # create persistent volumes, if any
+            if self.pvc_templates:
+                self.log.info("Create PVC")
+                pvc_results = await asyncio.gather(
+                    *(self._run_in_executor(
+                        v1.create_namespaced_persistent_volume_claim,
+                        self.namespace, self.make_pvc_spec(template))
+                      for template in self.pvc_templates), return_exceptions=True)
+
+            # TODO: check PVC creation results. ApiExceptions caused by
```

Comment:
Yeah, I think we need to individually check for PVCs existing. See https://github.com/jupyterhub/kubespawner/blob/master/kubespawner/spawner.py#L1760 for how this could possibly be?

Reply:
We might also wanna skip using asyncio.gather here, to make exception handling simpler? No issues otherwise though.